cpp/test/spatial/ball_cover.cu

```diff
@@ -226,11 +198,16 @@ class BallCoverAllKNNTest : public ::testing::TestWithParam<BallCoverInputs> {
                         d_train_inputs.data(), ToRadians());
     }
 
+    cudaStream_t *int_streams = nullptr;
+    std::vector<int64_t> *translations = nullptr;
+
     std::vector<float *> input_vec = {d_train_inputs.data()};
     std::vector<uint32_t> sizes_vec = {n};
 
-    compute_bfknn(handle, d_train_inputs.data(), d_train_inputs.data(), n, d, k,
-                  metric, d_ref_D.data(), d_ref_I.data());
+    raft::spatial::knn::detail::brute_force_knn_impl<uint32_t, int64_t>(
```

Comment:
Now that we have a knn that's not just proxying down to faiss, we should be gtesting it accordingly, similar to what's being done w/ the haversine and ball cover gtests. It's also important going forward because RAFT is beginning to get used by more projects and thus the impact of breaking tests is more than just cuml.

My suggestion is to test `l2_unexpanded_knn` and `l2_expanded_knn` directly in the gtests and then we can test the `brute_force_knn` more generally.

Reply:
certainly a rigorous tests within RAFT are needed for them, I'll add gtests for them separately, now instead of `l2_unexpanded_knn` and `l2_expanded_knn` we have single entry `fusedL2Knn` function for both of them.
For these kernel I relied on cuML cpp knn tests & pytests so far which is no longer correct as you rightly mention it.

Reply:
a new test `tests\spatial\fused_l2_knn.cu` is added which does testing for both L2 exp/unexp cases which compares its output with faiss bfknn call. 
I've also polished the fp32 atomicMax device function which I believe is more faster than atomicCAS based version and also takes care of NaNs.

Reply:
Apologies for the delay in updating this PR with the unit test